rectools/models/nn/transformer_base.py

```diff
@@ -682,19 +722,22 @@ def _init_data_preparator(self) -> None:
         raise NotImplementedError()
 
     def _init_trainer(self) -> None:
-        self._trainer = Trainer(
-            max_epochs=self.epochs,
-            min_epochs=self.epochs,
-            deterministic=self.deterministic,
-            enable_progress_bar=self.verbose > 0,
-            enable_model_summary=self.verbose > 0,
-            logger=self.verbose > 0,
-            enable_checkpointing=False,
-            devices=1,
-        )
+        if self.get_trainer_func is None:
+            self._trainer = Trainer(
+                max_epochs=self.epochs,
+                min_epochs=self.epochs,
+                deterministic=self.deterministic,
+                enable_progress_bar=self.verbose > 0,
+                enable_model_summary=self.verbose > 0,
+                logger=self.verbose > 0,
+                enable_checkpointing=self.verbose > 99,
+                devices=1,
+            )
+        else:
+            self._trainer = self.get_trainer_func()
```

Comment:
Let's maybe allow this func to accept smth like `context` argument where we can put all the model params?

It's a common practice. Saves user from redefining this func for every model

Reply:
If it's too complicated, then it's fine to keep as is

Reply:
Too complicated for configs. I would leave it as it is.

Reply:
I think there is a misunderstanding

Let's discuss on a call

Reply:
not going to do